lib/event_sourcery/event_processing/esp_runner.rb

```diff
@@ -34,25 +39,36 @@ def start!
         exit_indicating_status_of_processes
       end
 
+      def start_processor(event_processor)
+        process = ESPProcess.new(
+          event_processor: event_processor,
+          event_source: @event_source,
+          after_fork: @after_fork,
+        )
+        pid = Process.fork { process.start }
+        @pids[pid] = event_processor
+      end
+
+      def shutdown
+        @shutdown_requested = true
+      end
+
+      def shutdown_requested?
+        @shutdown_requested
+      end
```

Comment:
These three methods have been made public to provide options for responding to event processor failure.